File: benchmarks/comprehensive/javascript/recursive_tree.js
function treeSum(depth, value) {
    if (depth === 0) {
        return value;
    }
    
    // Binary tree recursion
    const left = treeSum(depth - 1, value * 2);
    const right = treeSum(depth - 1, value * 2 + 1);
    
    return left + right + value;
}

const DEPTH = 15;
const startTime = Date.now();
const result = treeSum(DEPTH, 1);
const endTime = Date.now();

console.log(`Tree sum (depth ${DEPTH}): ${result}`);
console.log(`Time: ${endTime - startTime}ms`);

File: benchmarks/comprehensive/python/recursive_tree.py
import time

def tree_sum(depth, value):
    if depth == 0:
        return value
    
    # Binary tree recursion
    left = tree_sum(depth - 1, value * 2)
    right = tree_sum(depth - 1, value * 2 + 1)
    
    return left + right + value

DEPTH = 15
start_time = time.time()
result = tree_sum(DEPTH, 1)
end_time = time.time()

print(f"Tree sum (depth {DEPTH}): {result}")
print(f"Time: {(end_time - start_time) * 1000:.2f}ms")